### ISSUE TITLE
gozip: Bad CPU type in executable

### ISSUE THREAD

**Issue by mxschmitt** [Azure/azure-functions-core-tools#3648]
When I try to deploy an app:

```
func azure functionapp publish <my-project> --javascript
```

I get:

```
Setting Functions site property 'netFrameworkVersion' to 'v8.0'
Getting site publishing info...
[2024-04-23T21:11:22.120Z] Starting the function app deployment...
Creating archive for current directory...
Creating archive for current directory...
Creating archive for current directory...
An error occurred trying to start process '/opt/homebrew/Cellar/azure-functions-core-tools@4/4.0.5611/gozip' with working directory '<my-project>'. Bad CPU type in executable
```

**Notes**

macOS 14.4.1 Silicon
Note: Rosetta emulation is **not** enabled

**Comment by fschlegelone** [comment]
Great that you made a PR to fix it, but why is nobody accepting it?

**Comment by drew1326** [comment]
+1 issue encountered on M2 Mac - please accept the fix!

**Comment by mxschmitt** [comment]
I can verify that its fixed for me on 4.0.6280.
